Dashboard/app/js/lib/views/devices/assignment-edit-views.jsx

```diff
@@ -731,5 +724,31 @@ FLOW.AssignmentEditView = FLOW.ReactComponentView.extend(
 
       this.renderReactSide();
     },
+
+    removeDatapointsFromAssignments(datapoints, deviceIdInString) {
+      // convert devieId to number
+      const deviceId = parseInt(deviceIdInString, 10);
+
+      // get the selected datapoint assignment
+      this.datapointAssignments = this.datapointAssignments.map(dpAssignment => {
+        // get the datapoints in the selected assignment
+        if (dpAssignment.deviceId !== deviceId) {
+          return dpAssignment;
+        }
+
+        // remove the selected datapoints from the list
+        const dps = dpAssignment.datapoints.filter(dp => {
+          // filter out datapoints that's in the array
+          return !datapoints.includes(dp.id);
+        });
+
+        return {
+          ...dpAssignment,
+          datapoints: dps,
+        };
+      });
```

Comment:
This section feels a bit confusing the way its constructed. The focus is on the single datapoint assignment that we have to update to reflect that its list of datapointIds has changed.  I dont think its necessary to go filtering through all the other datapoint assignments since all we do is just return as long as they dont match the one we want.